### PR TITLE
go: bump golang to 1.12.0

### PR DESCRIPTION
**Pull request by aanm**
Fix https://github.com/cilium/cilium/issues/7262

Fix https://github.com/cilium/cilium/issues/7261

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7211)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ dist: trusty
 sudo: required
 
 go:
- - 1.11.5
+ - 1.12.0
```
Comment (ianvernon) [inline .travis.yml]: Looks like Travis doesn't support go 1.12.0 yet :(

```
0.61s$ travis_setup_go
I don't have any idea what to do with '1.12.0'.
  (using download type 'auto')
Failed to run gimme
```

File: Dockerfile.builder [hunk 2]
```diff
@@ -14,7 +14,7 @@ WORKDIR /go/src/github.com/cilium/cilium
 ENV GOROOT /usr/local/go
 ENV GOPATH /go
 ENV PATH "$GOROOT/bin:$GOPATH/bin:$PATH"
-ENV GO_VERSION 1.11.5
+ENV GO_VERSION 1.12
```
Comment (ianvernon) [inline Dockerfile.builder:17]: will `1.12` always resolve to `1.12.0` ?

File: Makefile [hunk 3]
```diff
@@ -270,7 +270,7 @@ gofmt:
 
 govet:
 	@$(ECHO_CHECK) vetting all GOFILES...
-	$(QUIET)$(GO) tool vet api pkg test $(SUBDIRS)
```
Comment (ianvernon) [inline Makefile:276]: Please do two things:
* Comment out the preceding line as well
* Create an issue to track re-enabling of `go vet` with go 1.12 and refer to it in the `Makefile` with a comment.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Dockerfile.builder (hunk 2 above)
Comment (aanm) [reply Dockerfile.builder:17]: 1.12.0 does not exist, only 1.12, when 1.12.1 is released then 1.12.1 will be added

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: Makefile [hunk 4]
```diff
@@ -272,8 +272,9 @@ gofmt:
 	$(QUIET)for pkg in $(GOFILES); do $(GO) fmt $$pkg; done
 
 govet:
-	@$(ECHO_CHECK) vetting all GOFILES...
-	$(QUIET)$(GO) tool vet api pkg test $(SUBDIRS)
+	#FIXME https://github.com/cilium/cilium/issues/7261
```
Comment (tgraf) [inline Makefile]: What is the motivation to merge this before we fix 7261?

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

File: Makefile [hunk 5]
```diff
@@ -273,7 +274,25 @@ gofmt:
 
 govet:
 	@$(ECHO_CHECK) vetting all GOFILES...
-	$(QUIET)$(GO) tool vet api pkg test $(SUBDIRS)
+	time $(QUIET)$(GO) vet \
```
Comment (aanm) [inline Makefile]: I forgot to remove this time, I'll remove it once the tests have ran.

**Comment by aanm** [comment]
- tests have passed for all k8s versions
- upgrade test have failed for k8s 1.8, re-triggering without all k8s version

**Comment by aanm** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Pull request merged**